### PR TITLE
Add ApiProperties support for Cognitive Services

### PR DESCRIPTION
**Pull request by yangyuan**
## Description

* Add `New-AzCognitiveServicesAccountApiProperties` command.
* Support "ApiProperties" parameter for `New-AzCognitiveServicesAccount` and `Set-AzCognitiveServicesAccount`

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@yangyuan Please fix the issue report in static analysis

```
  D:\a\1\s\artifacts//StaticAnalysisResults\SignatureIssues.csv Errors
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.NewAzureCognitiveServicesAccountCommand","New-AzCognitiveServicesAccount","1","8410","Parameter ApiProperties of cmdlet New-AzCognitiveServicesAccount does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.NewAzureCognitiveServicesAccountApiPropertiesCommand","New-AzCognitiveServicesAccountApiProperties","1","8400","New-AzCognitiveServicesAccountApiProperties uses the noun 'AzCognitiveServicesAccountApiProperties', which does not follow the enforced naming convention of using a singular noun for a cmdlet name.","Consider using a singular noun for the cmdlet name."
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.SetAzureCognitiveServicesAccountCommand","Set-AzCognitiveServicesAccount","1","8410","Parameter ApiProperties of cmdlet Set-AzCognitiveServicesAccount does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
  
  
  D:\a\1\s\artifacts//StaticAnalysisResults\SignatureIssues.csv:
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.NewAzureCognitiveServicesAccountCommand","New-AzCognitiveServicesAccount","1","8410","Parameter ApiProperties of cmdlet New-AzCognitiveServicesAccount does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.NewAzureCognitiveServicesAccountApiPropertiesCommand","New-AzCognitiveServicesAccountApiProperties","1","8400","New-AzCognitiveServicesAccountApiProperties uses the noun 'AzCognitiveServicesAccountApiProperties', which does not follow the enforced naming convention of using a singular noun for a cmdlet name.","Consider using a singular noun for the cmdlet name."
  "D:\a\1\s\artifacts\Debug\Az.CognitiveServices\Microsoft.Azure.PowerShell.Cmdlets.CognitiveServices.dll","Microsoft.Azure.Commands.Management.CognitiveServices.SetAzureCognitiveServicesAccountCommand","Set-AzCognitiveServicesAccount","1","8410","Parameter ApiProperties of cmdlet Set-AzCognitiveServicesAccount does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
```

File: src/CognitiveServices/CognitiveServices/Az.CognitiveServices.psd1 [hunk 1]
```diff
@@ -86,7 +86,8 @@ CmdletsToExport = 'Get-AzCognitiveServicesAccount',
                'Get-AzCognitiveServicesAccountNetworkRuleSet', 
                'Update-AzCognitiveServicesAccountNetworkRuleSet', 
                'Add-AzCognitiveServicesAccountNetworkRule', 
-               'Remove-AzCognitiveServicesAccountNetworkRule'
+               'Remove-AzCognitiveServicesAccountNetworkRule', 
+               'New-AzCognitiveServicesAccountApiProperties'
```
Comment (msJinLei) [inline src/CognitiveServices/CognitiveServices/Az.CognitiveServices.psd1:90]: You should use singular for cmdlet name

File: src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/SetAzureCognitiveServicesAccount.cs [hunk 2]
```diff
@@ -145,6 +145,11 @@ public string KeyVaultUri
         [ValidateSet("Enabled", "Disabled", IgnoreCase = true)]
         public string PublicNetworkAccess { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The ApiProperties of Cognitive Services Account. Required by specific account types.")]
+        public CognitiveServicesAccountApiProperties ApiProperties { get; set; }
```
Comment (msJinLei) [inline src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/SetAzureCognitiveServicesAccount.cs:151]: Please use singular name ApiProperty

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by yangyuan** [comment]
Hi @msJinLei sorry for late action. I just updated the PR and now cmdlet and parameters are singular now.

File: src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesAccountApiProperty.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    /// Generate Cognitive Services Account ApiProperties class
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesAccountApiProperty", SupportsShouldProcess = true), OutputType(typeof(CognitiveServicesModels.CognitiveServicesAccountApiProperties))]
+    public class NewAzureCognitiveServicesAccountApiPropertiesCommand : CognitiveServicesAccountBaseCmdlet
+    {
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            WriteObject(new CognitiveServicesAccountApiProperties());
```
Comment (msJinLei) [inline src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesAccountApiProperty.cs:31]: if the cmdlet just create a local object, no need to support should process here.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by yangyuan** [comment]
After removing `should process` I got error says `New-AzCognitiveServicesAccountApiProperty Does not support ShouldProcess but the cmdlet verb New indicates that it should.` any suggestions?

**Comment by msJinLei** [comment]
> After removing `should process` I got error says `New-AzCognitiveServicesAccountApiProperty Does not support ShouldProcess but the cmdlet verb New indicates that it should.` any suggestions?

Please suppress it https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md#signature-issues

**Review by msJinLei — APPROVED** [review]

**Comment by yangyuan** [comment]
@msJinLei looks all passed :)

**Pull request merged**